### PR TITLE
Boost: Update dismissed alerts store

### LINKED ISSUE #33402: Boost: Update dismissed score prompt to use dismissed alerts data store
Use Dismissed Alerts DS introduced in #33133.

### PR DESCRIPTION
**Pull request by haqadn**
Fixes #33402

## Proposed changes:
* Use a combined `dismissed_alerts` store to handle score prompt dismissals.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
No

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Test if the previous state of speed score dismissals is kept in this version.
* Test that the dismissals work and PopOut is displayed without any issues.

You will have to rig the speed scores to test this. For that I usually manually [change this line](https://github.com/Automattic/jetpack/blob/d39cf216407f688b6111595e0e628c11e0819f8b/projects/packages/boost-speed-score/src/class-speed-score.php#L312) to something like:

```php
		$response = array(
			'status' => 'success',
			'scores' => [
				'current' => ['mobile' => 80, 'desktop' => 80],
				'noBoost' => ['mobile' => 100, 'desktop' => 100],
				'isStale' => false,
			]
		);
		return rest_ensure_response( $response );
```

### CODE REVIEW

**Comment by dilirity** [comment]
I can't get the "score increased" popup to show. I only get the "score decreased" popup.

File: projects/plugins/boost/compatibility/score-prompt.php [hunk 1]
```diff
@@ -4,11 +4,19 @@
...
  */
-
-$old_value = get_option( 'jb_show_score_prompt' );
+// Old value is the previous DS key, fallback to even older non-ds value.
+$old_value = get_option( 'jetpack_boost_ds_dismissed_score_prompt', get_option( 'jb_show_score_prompt' ) );
 if ( false !== $old_value ) {
-	$ds_value = get_option( 'jetpack_boost_ds_dismissed_score_prompt' );
-	if ( false === $ds_value ) {
-		add_option( 'jetpack_boost_ds_dismissed_score_prompt', $old_value );
+	$new_value = get_option( 'jetpack_boost_ds_dismissed_alerts' );
```
Comment (dilirity) [inline projects/plugins/boost/compatibility/score-prompt.php]: If there are no dismissed alerts when this runs, `get_option` should return `false`.

File: projects/plugins/boost/compatibility/score-prompt.php [hunk 2]
```diff
@@ -4,11 +4,19 @@
  *
  * @package automattic/jetpack-boost
  */
-
-$old_value = get_option( 'jb_show_score_prompt' );
+// Old value is the previous DS key, fallback to even older non-ds value.
+$old_value = get_option( 'jetpack_boost_ds_dismissed_score_prompt', get_option( 'jb_show_score_prompt' ) );
```
Comment (dilirity) [inline projects/plugins/boost/compatibility/score-prompt.php:8]: I love this!

File: projects/plugins/boost/app/assets/src/js/features/speed-score/speed-score.tsx [hunk 3]
```diff
@@ -136,7 +132,7 @@ const SpeedScore = () => {
 				{ siteIsOnline && <PerformanceHistory /> }
 			</div>
 
-			<PopOut scoreChange={ showScoreChangePopOut } onClose={ () => setClosePopOut( true ) } />
+			<PopOut scoreChange={ showScoreChangePopOut } />
```
Comment (dilirity) [inline projects/plugins/boost/app/assets/src/js/features/speed-score/speed-score.tsx:140]: Nice thinking moving the state to the popup itself!

**Review by dilirity — CHANGES_REQUESTED** [review]
Awesome work on this!

One small change and I couldn't get the "score increased" popup to show. I only get the "score decreased" popup.

After that, it should be ready for merge!

File: projects/plugins/boost/compatibility/score-prompt.php (hunk 1 above)
Comment (haqadn) [reply projects/plugins/boost/compatibility/score-prompt.php]: Addressed in a03afbb

**Comment by haqadn** [comment]
> Awesome work on this!
> 
> One small change and I couldn't get the "score increased" popup to show. I only get the "score decreased" popup.
> 
> After that, it should be ready for merge!

It does show up for me. Try this plugin to mock speed scores and make sure you have not previously disabled the alert.

[e2e-mock-speed-score-api.zip](https://github.com/Automattic/jetpack/files/13972789/e2e-mock-speed-score-api.zip)

**Review by dilirity — APPROVED** [review]
Thanks for the plugin! I got it to show.

Thanks to the plugin I discovered a bug that isn't caused by this PR. Let's merge this, and we'll address the bug in another PR.

Here's the ticket for the bug - https://github.com/Automattic/jetpack/issues/35102

**Pull request merged**